DEVELOPMENT.md

```diff
@@ -88,24 +88,26 @@ docker login $REGISTRY
 Here are some preliminary instructions. This still needs a lot of TLC. Feel free to send in PRs.
 
 ```shell
+# Start minikube
 minikube start
-kubectl create namespace eunomia
-kubectl apply -f ./deploy/crds/eunomia_v1alpha1_gitopsconfig_crd.yaml -n eunomia
-kubectl delete configmap eunomia-templates -n eunomia
-kubectl create configmap eunomia-templates --from-file=./templates/cronjob.yaml --from-file=./templates/job.yaml -n eunomia
-kubectl apply -f ./deploy/kubernetes -n eunomia
+
+# Deploy the operator pre-requisites, which require cluster-admin access
+helm template deploy/helm/prereqs/ | kubectl apply -f -
+
+# Deploy the operator
+helm template deploy/helm/operator/ | kubectl apply -f -
 ```
 
 ### Using Openshift
 
 Here are some preliminary instructions. This still needs a lot of TLC. Feel free to send in PRs.
 
 ```shell
-oc create namespace eunomia
-oc apply -f ./deploy/crds/gitops_v1alpha1_gitopsconfig_crd.yaml -n eunomia
-oc delete configmap eunomia-templates -n eunomia
-oc create configmap eunomia-templates --from-file=./templates/cronjob.yaml --from-file=./templates/job.yaml -n eunomia
-oc apply -f ./deploy/kubernetes -f ./deploy/openshift -n eunomia
+# Deploy the operator pre-requisites, which require cluster-admin access
```

Comment:
it looks like we are creating two charts one for the pre-requisites and one for the actual operator.
I'd recommend creating an wrapper chart that includes those two charts and ensures that the pre-reqs are run first. This will make the user experience easier.
Looking at the pre-reqs charts, I see that it contains also some of the RBAC related manifests. I'm, curious to know the reason of this choice. Usually permissions and roles go in the main chart.

Reply:
@raffaelespazzoli it's separated by the tasks that need cluster-admin and those that don't. It came from working on the 'cluster' demo. This will allow us to have the operator manage itself, but it will only need admin level access within the namespace.

Reply:
@raffaelespazzoli I'll have to look into the wrapper chart.

Reply:
@raffaelespazzoli  I did my required reading... ;-)

We can certainly move this into sub-charts, but I'm not sure how much sense it makes. It would only make it simpler for people testing out the operator, because it's one less command to run (1 instead of 2) to get it installed.
At the end when our users are leveraging our helm charts, they'd most like use them as 2 separate charts due to the required access differences to maintain it.
How do others feel about this? @sanbornick @seanmalloy ?

Reply:
@raffaelespazzoli @seanmalloy @sanbornick now that the e2e PR is merged, I can finish this one up. What should we do about helm chart and the potential usage of sub-charts. I'm currently not in favor adding sub-charts, because we need to keep it separate for the automation anyway due to different levels of access that are required. The only benefit I see is one less step in the demo.